### PR TITLE
fix: remove disabled flags from bulk evaluation

### LINKED ISSUE #652: [BUG] Bulk API evaluates disabled flags
### Observed behavior

Calls to the bulk evaluation API include disabled flags.

The following error is also written to the console:

```
playground-flagd-1           | 2023-05-11T17:53:38.965Z debug   eval/json_evaluator.go:263      requested flag is disabled: new-color-scheme    {"requestID": "chein8gulbm2ltkfoi80", "component": "evaluator", "evaluator": "json"}
playground-flagd-1           | 2023-05-11T17:53:38.965Z error   eval/json_evaluator.go:142      bulk evaluation: key: new-color-scheme returned error: FLAG_DISABLED    {"requestID": "chein8gulbm2ltkfoi80", "component": "evaluator", "evaluator": "json"}
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/eval.(*JSONEvaluator).ResolveAllValues
playground-flagd-1           |  /workspace/core/pkg/eval/json_evaluator.go:142
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/flag-evaluation.(*FlagEvaluationService).ResolveAll
playground-flagd-1           |  /workspace/core/pkg/service/flag-evaluation/flag_evaluator.go:60
playground-flagd-1           | github.com/bufbuild/connect-go.NewUnaryHandler[...].func1
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-go@v1.7.0/handler.go:52
playground-flagd-1           | github.com/bufbuild/connect-opentelemetry-go.(*Interceptor).WrapUnary.func1
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-opentelemetry-go@v0.1.0/interceptor.go:145
playground-flagd-1           | github.com/bufbuild/connect-go.NewUnaryHandler[...].func2
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-go@v1.7.0/handler.go:76
playground-flagd-1           | github.com/bufbuild/connect-go.(*Handler).ServeHTTP
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-go@v1.7.0/handler.go:238
playground-flagd-1           | net/http.(*ServeMux).ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2500
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/middleware/metrics.Middleware.Handler.func1.1
playground-flagd-1           |  /workspace/core/pkg/service/middleware/metrics/http_metrics.go:105
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/middleware/metrics.Middleware.Measure
playground-flagd-1           |  /workspace/core/pkg/service/middleware/metrics/http_metrics.go:90
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/middleware/metrics.Middleware.Handler.func1
playground-flagd-1           |  /workspace/core/pkg/service/middleware/metrics/http_metrics.go:104
playground-flagd-1           | net/http.HandlerFunc.ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2122
playground-flagd-1           | github.com/rs/cors.(*Cors).Handler.func1
playground-flagd-1           |  /go/pkg/mod/github.com/rs/cors@v1.9.0/cors.go:236
playground-flagd-1           | net/http.HandlerFunc.ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2122
playground-flagd-1           | golang.org/x/net/http2/h2c.h2cHandler.ServeHTTP
playground-flagd-1           |  /go/pkg/mod/golang.org/x/net@v0.9.0/http2/h2c/h2c.go:125
playground-flagd-1           | net/http.serverHandler.ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2936
playground-flagd-1           | net/http.(*conn).serve
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:1995
```

### Expected Behavior

Disabled flags should be excluded from bulk evaluation.

### Steps to reproduce

Start flagd with the following flags:

```json
{
	"flags": {
		"new-welcome-message": {
			"state": "ENABLED",
			"defaultVariant": "on",
			"variants": {
				"on": true,
				"off": false
			}
		},
		"new-color-scheme": {
			"state": "DISABLED",
			"defaultVariant": "off",
			"variants": {
				"on": true,
				"off": false
			}
		}
	}
}
```

Now cURL the resolve all endpoint http://localhost:8013/schema.v1.Service/ResolveAll

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Fixes #652 & enhance tests to validate error sates 

Note - had to sync go deps and thus the go mod/sum changes

### CODE REVIEW

File: core/pkg/eval/json_evaluator.go [hunk 1]
```diff
@@ -122,6 +122,11 @@ func (je *JSONEvaluator) ResolveAllValues(ctx context.Context, reqID string, con
 	var err error
 	allFlags := je.store.GetAll()
 	for flagKey, flag := range allFlags {
+		if flag.State == Disabled {
```
Comment (beeme1mr) [inline core/pkg/eval/json_evaluator.go:125]: What do you think about filtering out disabled flags when we're unmarshing the flag config? That would also indirectly resolve [this issue](https://github.com/open-feature/flagd/pull/670).

Comment (toddbaert) [reply core/pkg/eval/json_evaluator.go:125]: :thinking:

Comment (toddbaert) [reply core/pkg/eval/json_evaluator.go:125]: I think it's important for disabled flags to act as if they don't exist, as I mentioned elsewhere... how exactly we do that doesn't matter as much to me.

Comment (Kavindu-Dodan) [reply core/pkg/eval/json_evaluator.go:125]: shall we do the following, let's agree on the fixes proposed through #670 & #672 . And then follow up on the proposed behavior change - `Ignore/not store Disabled flags as if they do not exist`

The reason is this change requires considerable amount of changes. Besides, the above fixes do not add new behavior and merely bug fixes (can go into a release)

Comment (Kavindu-Dodan) [reply core/pkg/eval/json_evaluator.go:125]: Created issue https://github.com/open-feature/flagd/issues/673 , I can work on this. Feel free to update/comment on the issue

**Review by toddbaert — APPROVED** [review]

File: core/pkg/eval/json_evaluator.go [hunk 2]
```diff
@@ -122,6 +122,11 @@ func (je *JSONEvaluator) ResolveAllValues(ctx context.Context, reqID string, con
...
+			// ignore evaluation of disabled flag
+			continue
```
Comment (toddbaert) [inline core/pkg/eval/json_evaluator.go:127]: Think we can add a simple test case for this?

Comment (Kavindu-Dodan) [reply core/pkg/eval/json_evaluator.go:127]: I added a validation to the exisiting test in the latest commit [1]. 

[1] - https://github.com/open-feature/flagd/pull/672/commits/eef88dbc2bd8dbb5aaf0d2362fa5711a05dc9151

Comment (toddbaert) [reply core/pkg/eval/json_evaluator.go:127]: perfect!

**Review by toddbaert — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**